instrumentedProfilingLibrary/Exercises/time_profiler_constructor_destructor.cpp

```diff
@@ -0,0 +1,69 @@
+#include<iostream>
+#include<chrono>
+#include<cmath>
+
+
+using namespace std;
+
+class time_profiler{
+	public:
+		time_profiler(int N = 0): codeBlockStartTime(chrono::high_resolution_clock::now()){
```

Comment:
The parameter `N` seems unused. Maybe you want to use the `codeBlock` variable for that purpose instead and have a human readable name associated with what was measured

Reply:
corrected the unused variables `N` and `codeBlock`. Initially I had thought 'N' was being used in the main function which  was not true.